src/lua/luaScript.c

```diff
@@ -108,47 +105,60 @@ void unescapeScript(char *data)
+        if (page >= MAX_SCRIPT_PAGES) {
+                pr_error("lua: invalid script index\r\n");
+                return SCRIPT_ADD_RESULT_FAIL;
+        }
 
-    /*
-     * Stop LUA if we are flashing its data.  This is mainly done to recover
-     * RAM since our flashing operation is a heavy bugger
-     */
-    terminate_lua();
+        switch (mode) {
+        case SCRIPT_ADD_MODE_IN_PROGRESS:
+        case SCRIPT_ADD_MODE_COMPLETE:
+                /* Valid cases.  Carry on */
+                break;
+        default:
+                pr_error_int_msg("Unknown script_add_mode: ", mode);
+                return SCRIPT_ADD_RESULT_FAIL;
+        }
 
-    if (page < MAX_SCRIPT_PAGES) {
-        if (mode == SCRIPT_ADD_MODE_IN_PROGRESS || mode == SCRIPT_ADD_MODE_COMPLETE) {
-            if (g_scriptBuffer == NULL) {
-                g_scriptBuffer = (ScriptConfig *)portMalloc(sizeof(ScriptConfig));
-                memcpy((void *)g_scriptBuffer, (void *)&g_scriptConfig, sizeof(ScriptConfig));
-            }
+        static ScriptConfig *g_scriptBuffer;
+        if (g_scriptBuffer == NULL) {
+                terminate_lua();
 
-            if (g_scriptBuffer != NULL) {
-                page = page * SCRIPT_PAGE_SIZE;
-                char *pageToAdd = g_scriptBuffer->script + page;
-                strncpy(pageToAdd, data, SCRIPT_PAGE_SIZE);
-
-                if (mode == SCRIPT_ADD_MODE_COMPLETE) {
-                    pr_info("lua: update complete, flashing: ");
-                    if (memory_flash_region((void *)&g_scriptConfig, (void *)g_scriptBuffer, sizeof(ScriptConfig)) == 0) {
-                        pr_info("win\r\n");
-                    } else {
-                        pr_error("fail\r\n");
-                        result = SCRIPT_ADD_RESULT_FAIL;
-                    }
-                    portFree(g_scriptBuffer);
-                    g_scriptBuffer = NULL;
-                }
-            } else {
-                pr_error("lua: script buffer alloc fail\r\n");
-                result = SCRIPT_ADD_RESULT_FAIL;
-            }
+                pr_info("Allocating new script buffer\r\n");
+                g_scriptBuffer =
+                        (ScriptConfig *) portMalloc(sizeof(ScriptConfig));
+                memcpy((void *)g_scriptBuffer, (void *)&g_scriptConfig,
+                       sizeof(ScriptConfig));
         }
-    } else {
-        pr_error("lua: invalid script index\r\n");
-        result = SCRIPT_ADD_RESULT_FAIL;
-    }
-    return result;
+
+        if (g_scriptBuffer == NULL) {
+                pr_error("Failed to allocate memory for script buffer.\r\n");
+                return SCRIPT_ADD_RESULT_FAIL;
+        }
+
+        char *pageToAdd = g_scriptBuffer->script + page * SCRIPT_PAGE_SIZE;
+        strncpy(pageToAdd, data, SCRIPT_PAGE_SIZE);
+
+        if (mode == SCRIPT_ADD_MODE_IN_PROGRESS)
+                return SCRIPT_ADD_RESULT_OK;
+
+        pr_info("Completed updating LUA. Flashing... ");
```

Comment:
should prefix with lua:

suggest:   "lua: completed updating script. Flashing... "